src/app/ui/TaskBoard.tsx

```diff
@@ -88,7 +88,7 @@ export const TaskBoard = ({ mode }: TaskBoardProps) => {
 
   const isNoTasksWithFilter = tasks && !userHasNoFilter && !filteredTasks.length
 
-  if (tasks && tasks.length === 0 && userHasNoFilter && !isTasksLoading) {
+  if (tasks && tasks.length === 0 && userHasNoFilter) {
     return (
```

Comment:
Is there a reason why we are removing `isTasksLoading` check? This would cause a regression for a previous fix. Originally it was intended to remove sideeffects so we don't show "No tasks found" when loading tasks

Right now "No tasks found" state occurs even though there are valid tasks in the following cases:
- When toggling show archived / show unarchived
- Brief flicker on loading tasks board for the first time (happens for a split second but is noticable sometimes like in 0:21 in screencast)

[Screencast from 2024-11-27 13-27-15.webm](https://github.com/user-attachments/assets/15c31c9d-20f2-406f-8d46-2b6f223d3626)